### PR TITLE
Refactor dataset tree tests

### PR DESCRIPTION
**Pull request by stepanzharychevbroadcom**
## Proposed changes

Refactoring of Unit Tests related to old Dataset Tree file to match actual standards of test implementation.

Milestone:
1.7 Release

Changelog:
Added DatasetTree tests to /dataset folder.
Removed old implementation of the same tests.

## Types of changes

What types of changes does your code introduce to Zowe Explorer?
_Put an `x` in the boxes that apply_

- [ ] Bugfix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [x] Updates to Documentation or Tests (if none of the other choices apply)

## Checklist

_Put an `x` in the boxes that apply. You can also fill these out after creating the PR. If you're unsure about any of them, don't hesitate to ask. We're here to help! This checklist will be used as reference for both the contributor and the reviewer_

- [x] I have read the [CONTRIBUTOR GUIDANCE](https://github.com/zowe/vscode-extension-for-zowe/wiki/Best-Practices:-Contributor-Guidance) wiki
- [x] PR title follows [Conventional Commits Guidelines](https://www.conventionalcommits.org/en/v1.0.0-beta.2/)
- [x] PR Description is included
- [x] `npm run vscode:prepublish` has been executed
- [x] All checks have passed (DCO, Jenkins and Code Coverage)
- [x] I have added unit test and it is passing
- [ ] I have added integration test and it is passing
- [x] There is coverage for the code that I have added
- [x] I have tested it manually and there are no regressions found
- [x] I have added necessary documentation (if appropriate)
- [x] Any PR dependencies have been merged and published (if appropriate)

## Further comments

If this is a relatively large or complex change, kick off the discussion by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

**Comment by jellypuno** [comment]
@stepanzharychevbroadcom @zdmullen Is this ready for review? if yes, can you fill out the PR template?

**Comment by zdmullen** [comment]
@stepanzharychevbroadcom, can you fill out the template so that it is clear what this PR is about and that it is ready to be reviewed? Thanks.

**Comment by stepanzharychevbroadcom** [comment]
@zdmullen @jellypuno Fixed.

**Comment by stepanzharychevbroadcom** [comment]
> Looks Good! You ticked integration as part of this PR and I don't see any.

@jellypuno I checked everything. Now it's unchecked.

File: __tests__/__unit__/utils.unit.test.ts [hunk 1]
```diff
@@ -0,0 +1,79 @@
...
+    Object.defineProperty(vscode.window, "showQuickPick", { value: jest.fn(), configurable: true });
+    Object.defineProperty(vscode.window, "createQuickPick", { value: jest.fn(), configurable: true });
+    Object.defineProperty(vscode.window, "showInputBox", { value: jest.fn(), configurable: true });
+    Object.defineProperty(vscode.window, "showErrorMessage", { value: jest.fn(), configurable: true });
+    Object.defineProperty(Profiles, "getInstance", { value: jest.fn(), configurable: true });
+    Object.defineProperty(globals, "ISTHEIA", { get: isTheia, configurable: true });
+
+    return {
+        isTheia
+    };
```
Comment (VitGottwald) [inline __tests__/__unit__/utils.unit.test.ts:30]: This is suspicious.  The createGlobalMocks() function operates by side effect, but here it also returns a value. We should separate these concerns.

Comment (VitGottwald) [inline __tests__/__unit__/utils.unit.test.ts:30]: I suggest we pass an argument to createGlobalMocks to set up the mocks properly like this:

```suggestion
function createGlobalMocks({isTheia=true} = {}) {
    Object.defineProperty(vscode.window, "showQuickPick", { value: jest.fn(), configurable: true });
    Object.defineProperty(vscode.window, "createQuickPick", { value: jest.fn(), configurable: true });
    Object.defineProperty(vscode.window, "showInputBox", { value: jest.fn(), configurable: true });
    Object.defineProperty(vscode.window, "showErrorMessage", { value: jest.fn(), configurable: true });
    Object.defineProperty(Profiles, "getInstance", { value: jest.fn(), configurable: true });
    Object.defineProperty(globals, "ISTHEIA", { value: isTheia, configurable: true });
    }
```

and that we then use it in tests like this

```js
        const globalMocks = createGlobalMocks({isTheia: true});
```

File: __tests__/__unit__/dataset/DatasetTree.unit.test.ts [hunk 2]
```diff
@@ -0,0 +1,1176 @@
...
+    createTreeView, createWorkspaceConfiguration
+} from "../../../__mocks__/mockCreators/shared";
+import { createDatasetSessionNode } from "../../../__mocks__/mockCreators/datasets";
+import { bindMvsApi, createMvsApi } from "../../../__mocks__/mockCreators/api";
+
+jest.mock("fs");
+jest.mock("util");
+
+function createGlobalMocks() {
+    const isTheia = jest.fn();
```
Comment (VitGottwald) [inline __tests__/__unit__/dataset/DatasetTree.unit.test.ts:34]: Same comment as I did in util test file. I guess this is a common pattern. Let's leave it in here for this PR and make a followup that will fix it across the codebase.

**Pull request merged**